### ISSUE TITLE
mme crash during multi-UE attach, unable to recover itself

### ISSUE THREAD

**Issue by jpad-freedomfi** [magma/magma#3864]
### Your Environment

- **Version:** v1.3.0-1602477016-723feee0 with patched MME binary
- **Affected Component:** Access Gateway
- **Affected Subcomponent:** MME
- **Deployment Environment:**  bare-metal (AGW)
- **EPC Configuration:** 2 APN resources (VLAN tagged), 2 APNs, static SGi_mgmt_iface IP, bridge/DHCP_Broadcast ip allocation. 

### Describe the Issue

We were installing AGW in a live site. We performed a cutover from Baicells cloud core to Magma AGW. This was done by rebooting eNB, while eNB is down, we change the router configuration. Once eNB comes back up, the UEs all attempt to attach fairly rapidly, attempt rate is not in our control.

After a couple of seconds, we observed no reply to _InitialUEMessage, Attach Request, PDN Connectivity Request_, which start piling up as MME is down and UEs keep retrying.

One other note, we did not know at the time, but many of the UEs were configured with APN names not configured in Magma. This resulted in fairly rapid failure and retry loops on the order of a few seconds (~5sec) per UE to retry which made the MME load bigger. In addition, some of the requested APN names had spaces. Not sure it matters, but I hadn't see that before. 

**Likely Steps To Reproduce**
1. Build 1.3.0-1602477016-723feee0
2. Replace MME binary with attached version
3. Configure a collection of UEs (we had 6 UEs) 2 with valid APNs configured, the other 4 or more with invalid APN names (possibly some with spaces?)
4. Have UEs all attempt to join with random or no staggering. 
5. Observe once the 4 UEs fail and retry, at some point MME will die. 

**Expected behavior**
Expected behavior is continued graceful rejection of bad APN requests until we could configure the right APN names and assign to subscriber IMSI. Also it is expected that if the MME dies, it should be able to restart on it's own. (There is chance we/I didn't wait long enough as this was a live site, but it did not seem to be restarting itself).

As this was a live site, and the APN issue wasn't expected, I didn't have a purposeful log collection going. Attached are the logs I was able to get after the fact. 
- 4 core dump files
- full journalctl log from boot to after the event (we eventually moved eNB back to CloudCore)
- /var/log/messages
- ovs-vswitchd.log
- all archived mme logs
- sctpd filtered journalctl logs
[AP06_lockup.zip](https://github.com/magma/magma/files/5626673/AP06_lockup.zip)
- mme binary used
[mme.zip](https://github.com/magma/magma/files/5626676/mme.zip)

**Comment by berezovskyi-oleksandr** [comment]
I took a look at the crashdumps and all crashes are at the same code:
```
Core was generated by `/usr/local/bin/mme -c /var/opt/magma/tmp/mme.conf -s /var/opt/magma/tmp/spgw.co'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x000055a6faab9970 in mme_app_handle_delete_session_rsp (mme_app_desc_p=0x60c000018d00, delete_sess_resp_pP=0x62900011822d) at /home/vagrant/magma/lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c:933
```
This line leads to the code:
```
    pdn_cid_t pid =
        ue_context_p->bearer_contexts[EBI_TO_INDEX(delete_sess_resp_pP->lbi)]
            ->pdn_cx_id;
```

Value of the `delete_sess_resp_pP->lbi` is 0. 
`EBI_TO_INDEX` is a definition, that decrements value by 5, and thus the index will became -5.
Getting a value by index below 0 leads to the segmentation fault. 

This code appeared at PR #2941

**Comment by ulaskozat** [comment]
@pruthvihebbani could you check if there are issues with PR #2941 that would cause a 0 value to be passed for  `delete_sess_resp_pP->lbi`

**Comment by berezovskyi-oleksandr** [comment]
@ulaskozat , wanted to mention, that there might be no problem with the PR itself. 

Before this PR, there were just no checks for `delete_sess_resp_pP->lbi`. So the problem may be in two different places:
- in the condition which PR #2941 introduced
- in the code somewhere else, which leads to `delete_sess_resp_pP->lbi=0`

**Comment by pruthvihebbani** [comment]
There is a problem in spgw not in PR #2941. lbi is not getting populated in delete_sess_resp_pP message if delete session req is not accepted. This is getting checked at mme app in PR #2941 which is leading to the crash

**Comment by jpad-freedomfi** [comment]
@prabinakpattnaik and @ulaskozat thank you for getting on this. We tried to cherry pick and use the latest 1.3.0 version plus your commit for this issue. It appears we don't have support for dual APN with the MME binary we made. Traffic on second APN is not being VLAN tagged as it should be. 

We will keep trying to find a version that supports both. 

@berezovskyi-oleksandr can provide the commit he used v1.3.0
